src/server/string_family.cc

```diff
@@ -180,7 +177,7 @@ OpResult<TResult<size_t>> OpStrLen(const OpArgs& op_args, string_view key) {
     fb2::Future<size_t> fut;
     op_args.shard->tiered_storage()->Read(
         op_args.db_cntx.db_index, key, co,
-        [fut](const std::string& s) mutable { fut.Resolve(s.size()); });
+        [fut](io::Result<std::string> s) mutable { fut.Resolve(s->size()); });
```

Comment:
doesn't that trigger a deep copy of the `io::Result` which will deep copy the string ?